internal/interactivity/multiInput.go

```diff
@@ -37,6 +37,7 @@ type MultiInput struct {
 	cursorMode cursor.Mode
 	focusIndex int
 	done       bool
+	signalExit bool
```

Comment:
Really don't think adding this to every model should be necessary. Can't we just put at the top of every `Update(msg tea.Msg)` func something like:
```
charm.CheckExit(msg)
```
Which calls (pseudocode):
```
func CheckExit(msg tea.Msg) {
	if msg.String() == "ctrl+c", "esc" {
		os.Exit(0)
	}
}
```
As far as I can tell that would work for every case here while simplifying the code (we can remove the `case "ctrl+c"` blocks everywhere, also enforces uniform exit commands) rather than making it more verbose

Reply:
The problem is you can't os exit from the Update function. That's not a graceful exit. The Run function needs to actually complete for charm's internal stuff to give up the cursor correctly.

Reply:
So we can factor out CheckExit, but you can't call os.Exit from it. You'll run into the same problem.